### DISCUSSION TITLE
Is there way to fitting text inside a shape?

### DISCUSSION THREAD

**Discussion by 0xlkda** [fabricjs/fabric.js#7458] [Q&A]
I want to put text inside circle, triangle, or heart shape.

Many thanks

**Comment by asturur** [comment]
You should look at text on a path, an you should first create a path with the shape you are interested in. Usually starting from an SVG.
Circle is a particular use case where you need recreate the circle with the path svg element rather than using the native circle.

**Reply by asturur** [reply]
Look for text.path on the docs.
There are a bunch of additional properties to configure the behaviour

**Comment by 0xlkda** [comment]
![image](https://user-images.githubusercontent.com/14184017/139204475-21c92603-b602-4329-9334-3321737e6b5a.png)
I want something like this. Seem that text.path cannot do this.

**Reply by asturur** [reply]
this can't be done automatically right now.
There is logic missing with path intersections.
We would need to be able to undertand, given a path and a line where do  those cross.
On top of that we would need to frame this functionality in a way that is useful in general for text.


In theory you can still do this defining a path made of a series of orizontal lines that have the hearth shape. Text will flow from a line to another automatically.
